ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedSessionTable.java

```diff
@@ -78,13 +69,19 @@
   }
 
   private List<WindowedRow> findSession(
-      final ReadOnlySessionStore<GenericKey, GenericRow> store,
       final GenericKey key,
       final Range<Instant> windowStart,
       final Range<Instant> windowEnd
   ) {
+
+    final WindowRangeQuery<GenericKey, GenericRow> query = WindowRangeQuery.withKey(key);
```

Comment:
I'm wondering why we cannot now pass in the `windowStart/End` as query parameters directly?

Reply:
yea..the old code didn't do it but manually processed the records so I assumed it to be safer to do it the same way. I suggest we make it the goal of this PR to create parity with the IQv1, and then add optimizations like this as small follow-up PRs.

Reply:
@guozhangwang agree that some of the old tests are still valuable, I added tests for iterator closing and empty sets.